### PR TITLE
Make CI script work with github runners

### PR DESCRIPTION
**Pull request by segeljakt**

### CODE REVIEW

**Review by frej — CHANGES_REQUESTED** [review]
`run-build` sets `A2M_CCACHE="1"`, it shouldn't do that when we're running without the `PERSIST_DIR` as it will probably write files to a sibdir of the user's home directory.

File: .github/continuous-integration.sh [hunk 1]
```diff
@@ -56,7 +60,11 @@ function check-ccache {
 }
 
 function run-build {
-    A2M_CCACHE="1" BUILD_FLAVOUR="Release" A2M_ASSERTS="1" run-step ./build
+    if [[ -z "${PERSIST_DIR}" ]]; then
+        BUILD_FLAVOUR="Release" A2M_ASSERTS="1" run-step ./build
```
Comment (frej) [inline .github/continuous-integration.sh:64]: Duplicating command lines are not a good thing, we should do it like we do it in arc, i.e.

```shell
A2M_CMAKE_DEFS=('BUILD_FLAVOUR="Release"' 'A2M_ASSERTS="1"')

if [[ ! -z "${PERSIST_DIR}" ]]; then
  A2M_CMAKE_DEFS+=('A2M_CCACHE="1"')
fi
...
"${A2M_CMAKE_DEFS[@]}" run-step ./build
```

Comment (segeljakt) [reply .github/continuous-integration.sh]: I updated the script now 👍

File: .github/continuous-integration.sh [hunk 2]
```diff
@@ -56,7 +60,13 @@ function check-ccache {
...
 
 function run-build {
-    A2M_CCACHE="1" BUILD_FLAVOUR="Release" A2M_ASSERTS="1" run-step ./build
+    A2M_CMAKE_DEFS=('BUILD_FLAVOUR="Release"' 'A2M_ASSERTS="1"')
+
+    if [[ ! -z "${PERSIST_DIR}" ]]; then
+        A2M_CMAKE_DEFS+=('A2M_CCACHE="1"')
+    fi
+
+    "${A2M_CMAKE_DEFS[@]}" run-step ./build
```
Comment (frej) [inline .github/continuous-integration.sh:69]: Sorry for that suggestion, we have to do `env "${A2M_CMAKE_DEFS[@]}" run-step ./build`for the environment variables to be set.

**Review by frej — APPROVED** [review]
If you fix my broken suggestion, the test will hopefully pass.

**Comment by frej** [comment]
```
env: 'run-step': No such file or directory
```
Ah, run-step is a bash function, we have to `export -f` it.

So add `export -f run-step` after the definition.

**Comment by frej** [comment]
Try this
```bash
env "${A2M_CMAKE_DEFS[@]}" bash -c run-step ./build
```

**Pull request closed without merging**